qiskit/pulse/timing.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Private utility methods used by Schedules."""
+
+from typing import List, Tuple
+
+from .exceptions import PulseError
+
+
+Interval = Tuple[int, int]
+
+
+def _insertion_index(intervals: List[Interval], new_interval: Interval, index: int = 0) -> int:
```

Comment:
should these be considered internal methods but still not have `_` if they are being imported into `schedule`? If you don't want to expose the recursive binary search, perhaps you could create another method `insertion_index` that calls `_insertion_index`?